src/common/cdn/index.js

```diff
@@ -0,0 +1,108 @@
+import { loader } from '../loader';
+import versions from './versions.json';
+const MAX_RETRIES = 3;
+
+/**
+ * @config object with the following (all are required)
+ * string key : the key to lookup both in versions and CDN url
+ * [] files : The files to lookup in the CDN
+ * [] types :  The Types of files for CDN loader (see common/loader)
+ * function setLoaded(boolean) : sets the is loaded state
+ * function handleSuccess() : when cdn is succsssfully loaded
+ * function handleError(err) : when the cdn loading fails
+ * boolean stagger :  (optional) if all promoses should be staggered or executed in any order
+ */
+export class CDNLoader {
+    constructor(self, { key, files, types, setLoading, handleSuccess, handleError, stagger }) {
+        this.self = self;
+        this.retryTimes = 0;
+        this.setLoading = setLoading;
+        this.handleError = handleError;
+        this.handleSuccess = handleSuccess;
+        this.files = files;
+        this.types = types;
+        this.key = key;
+        this.stagger = stagger;
+        this._setFiles();
+    }
+
+    setOverrides(overrides, version) {
+        this._setFiles(overrides, version);
+
+        return this;
+    }
+
+    _setFiles(overrides, version) {
+        this.cdnFiles = [];
+        this.url = `https://ir.ebaystatic.com/cr/v/c1/ebayui/${this.key}/v${
+            version || versions[this.key]
+        }`;
+        for (let i = 0; i < this.files.length; i++) {
+            if (overrides && overrides[i]) {
+                this.cdnFiles[i] = overrides[i];
+            } else {
+                this.cdnFiles[i] = `${this.url}/${this.files[i]}`;
+            }
+        }
+    }
+
+    mount() {
+        this.isLoaded = false;
+
+        if (document.readyState === 'complete') {
+            this.loadCDN();
+        } else {
+            this.self.subscribeTo(window).once('load', this.loadCDN.bind(this));
+        }
+    }
+
+    loadCDN(immediate) {
+        const _timeout =
+            window.requestIdleCallback ||
+            function (handler, arg) {
+                return setTimeout(() => {
+                    handler();
+                }, arg.timeout);
+            };
+
+        const _cancel =
+            window.cancelIdleCallback ||
+            function (id) {
+                clearTimeout(id);
+            };
+
+        this.retryTimes = 0;
+        this.setLoading(false);
+
+        _cancel(this.loadDelay);
+        if (!immediate) {
+            this.setLoading(false);
```

Comment:
Do we need this line? We already set it at L75. Also, I guess we need to set loading to true in L80 callback.

Reply:
I'm guessing one of these needed to be `true` and the other sets it to the opposite based on the condition.

Reply:
This is actually correct. If we trigger the immediate loading, we want to make sure the setLoading is set to false. This is if multiple components are on the page running concurrently.